charts/gateway-helm/values.tmpl.yaml

```diff
@@ -93,6 +93,10 @@ hpa:
   behavior: {}
 
 config:
+  deploy:
+    # Deploy type controls which namespace EnvoyGateway will povison into.
+    # Optional value: ControllerNamespace, GatewayNamespace
```

Comment:
this should be inside provider.kubernetes

Reply:
I want to add a knob won't be part of configmap, so that users can enabled rbac first and reenable GNM later.

Reply:
users will still need to then define twice, and the primary way to define configimap is via helm today

Reply:
either way is fine in this patch:
```
{{ if or $enabledInConfig $enabledInConfigmap }}
```